VAMobile/src/store/slices/authSlice.ts

```diff
@@ -521,6 +521,9 @@ export const logout = (): AppThunk => async (dispatch, getState) => {
     await clearStoredAuthCreds()
     api.setAccessToken(undefined)
     api.setRefreshToken(undefined)
+    // Remove SID to trigger device registration with push service on next login. This ensures
+    // future logins won't receive push notifications for the user that was previously logged-in.
+    await AsyncStorage.removeItem(DEVICE_ENDPOINT_SID)
```

Comment:
doesn't this just unregisters the device locally?

Sure the new user would be newly registered with this device, but wouldn't the old user still be registered with this device since we pass those old details to the BE with /v0/push/register. We don't clear what the BE has when someone logs out is what I am getting at and that is the primary thing that controls push notifications reaching the device.